File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    string s = to_string(x);
    int num = 0;
    for (int i = 0; i < s.size(); i++) {
        num += (s[i]-'0');
    }
    if (x % num != 0) {
        answer = false;
    }
    return answer;
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
import java.util.*;

class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        String s = String.valueOf(x);
        int sum = Arrays.stream(s.split("")).mapToInt(Integer::parseInt).sum();
        answer = x % sum == 0;
        return answer;
    }
}
